File: 2022/01/p.cc
#include <bits/stdc++.h>

int main() {
    std::vector<int> sums;
    for (;;) {
        int sum = 0, cnt = 0;
        std::string line;
        while (std::getline(std::cin, line) && !line.empty()) {
            sum += std::stoi(line);
            cnt++;
        }
        if (cnt == 0) {
            break;
        }
        sums.push_back(sum);
    }

    std::sort(sums.begin(), sums.end());
    std::reverse(sums.begin(), sums.end());

    std::cout << sums[0] << "\n" << sums[0] + sums[1] + sums[2] << "\n";
}

File: 2022/01/p.go
package main

import (
	"fmt"
	"sort"
)

func main() {
	sums := []int{}
	for {
		sum, cnt := 0, 0
		for {
			var x int
			if k, _ := fmt.Scanln(&x); k == 0 {
				break
			}
			sum += x
			cnt += 1
		}
		if cnt == 0 {
			break
		}
		sums = append(sums, sum)
	}

	sort.Sort(sort.Reverse(sort.IntSlice(sums)))
	fmt.Println(sums[0], sums[0]+sums[1]+sums[2])
}
